pkg/pci/function.go

```diff
@@ -0,0 +1,109 @@
+//
+// Use and distribution licensed under the Apache license version 2.
+//
+// See the COPYING file in the root project directory for full text.
+//
+
+package pci
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/jaypipes/ghw/pkg/context"
+	"github.com/jaypipes/ghw/pkg/marshal"
+)
+
+// Function describes an SR-IOV physical or virtual function. Physical functions
+// will have no Parent Function struct pointer and will have one or more Function
+// structs in the Functions field.
+type Function struct {
+	Device // All Functions are PCI Devices
+	// Parent contains a pointer to the parent physical function.
+	// Will be empty when this is a physical function
+	Parent *Function `json:"parent,omitempty"`
+	// MaxVFs contains the maximum number of supported virtual
+	// functions for this physical function
+	MaxVFs int `json:"max_vfs"`
+	// Functions contains the physical function's virtual functions
+	Functions []*Function `json:"functions"`
```

Comment:
Why not just add these attributes to `pkg/pci.Device`?

Reply:
on it!